### PR TITLE
Accept ADVERTISE and NODE env vars

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
It's obvious that this needs to accept cli flags instead of having so many ENV vars (and the tests need to be reworked to be more granular), but I think that needs to be a separate refactoring.

File: consul-join.go [hunk 1]
```diff
@@ -58,9 +58,17 @@ func bootstrapExpectFromLookup(s string, exists bool) int64 {
 	return 3
 }
 
-func fullCommandArgs(joinArgs []string, bootstrapExpect int64, osArgs []string) []string {
+func fullCommandArgs(joinArgs []string, node string, advertise string, bootstrapExpect int64, osArgs []string) []string {
 	commandArgs := []string{consul, "agent"}
 
+	if node != "" {
```
Comment (rylev) [inline consul-join.go:64]: so if node is `""` then we just leave that off?

Comment (ghost) [inline consul-join.go:64]: Yes, it will be "" if the ENV var is empty and consul will use $HOSTNAME if it's missing.

**Comment by rylev** [comment]
LGTM

**Pull request merged**